lib/jekyll/filters/url_filters.rb

```diff
@@ -20,13 +20,17 @@ def absolute_url(input)
         ).normalize.to_s
       end
 
-      # Produces a URL relative to the domain root based on site.baseurl.
+      # Produces a URL relative to the domain root based on site.baseurl
+      # unless it is already an absolute url with an authority (host).
```

Comment:
small correction here. **`authority != host`**
```ruby
def uri(input = "http://localhost:4000")
  Addressable::URI.parse(input.to_s)
end

print uri.absolute? # => true
print uri.host      # => "localhost"
print uri.authority # => "localhost:4000"
```

Reply:
It's not supposed to mean authority == host but host is the integral component of an authority. No host, no authority. It may also have userinfo and port, but they're not the key.

Reply:
okay then..